spec/functions/deprecation_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'spec_helper'
+require "puppet_x/puppetlabs/stdlib/deprecation_state"
+
+if ENV["FUTURE_PARSER"] == 'yes'
+  describe 'deprecation' do
+    pending 'teach rspec-puppet to load future-only functions under 3.7.5' do
+      it { is_expected.not_to eq(nil) }
+    end
+  end
+end
+
+if Puppet.version.to_f >= 4.0
+  describe 'deprecation' do
+    before(:each) {
+      # this is to reset the deprecation warning track, and strict variable to default
+      PuppetX::Puppetlabs::Stdlib::DeprecationState.class_variable_set :@@deprecation_warning_said, []
+      Puppet.settings[:strict] = :warning
```

Comment:
Affecting global state should be reversed in `after` block, to avoid poisoning the environment for later tests.